src/liboslexec/llvm_gen.cpp

```diff
@@ -610,19 +610,28 @@ LLVMGEN (llvm_gen_div)
 
     // The following should handle f/f, v/v, v/f, f/v, i/i
     // That's all that should be allowed by oslc.
+    const char *safe_div = is_float ? "osl_safe_div_fff" : "osl_safe_div_iii";
     bool deriv = (Result.has_derivs() && (A.has_derivs() || B.has_derivs()));
     for (int i = 0; i < num_components; i++) {
         llvm::Value *a = rop.llvm_load_value (A, 0, i, type);
         llvm::Value *b = rop.llvm_load_value (B, 0, i, type);
         if (!a || !b)
             return false;
-        llvm::Value *a_div_b = rop.ll.op_make_safe_div (type, a, b);
+        llvm::Value *a_div_b;
+        if (B.is_constant() && ! rop.is_zero(B))
+            a_div_b = rop.ll.op_div (a, b);
+        else
+            a_div_b = rop.ll.call_function (safe_div, a, b);
         llvm::Value *rx = NULL, *ry = NULL;
 
         if (deriv) {
             // Division of duals: (a/b, 1/b*(ax-a/b*bx), 1/b*(ay-a/b*by))
             ASSERT (is_float);
-            llvm::Value *binv = rop.ll.op_make_safe_div (type, rop.ll.constant(1.0f), b);
+            llvm::Value *binv;
+            if (B.is_constant() && ! rop.is_zero(B))
+                binv = rop.ll.op_div (rop.ll.constant(1.0f), b);
+            else
+                binv = rop.ll.call_function (safe_div, rop.ll.constant(1.0f), b);
```

Comment:
If the denominator is a nonzero constant, it will just do a plain div. Otherwise, it will call the safe_div function that tests for zero, etc. That function itself will generally be inlined by the time LLVM optimization is done. It's just easier to call that inlineable function rather than fiddle with making conditionals and blocks here.